frontend/public/extend/devconsole/utils/pipeline-actions.tsx

```diff
@@ -0,0 +1,207 @@
+          : pipeline.metadata.namespace || '',
+      labels:
+        latestRun && latestRun.metadata && latestRun.metadata.labels
+          ? latestRun.metadata.labels
+          : {
+            'tekton.dev/pipeline': pipeline.metadata.name,
+          },
+    },
+    spec: {
+      pipelineRef: {
+        name:
+          latestRun &&
+          latestRun.spec &&
+          latestRun.spec.pipelineRef &&
+          latestRun.spec.pipelineRef.name
+            ? latestRun.spec.pipelineRef.name
+            : pipeline && pipeline.metadata && pipeline.metadata.name
+              ? pipeline.metadata.name
+              : null,
+      },
+      params:
+        latestRun && latestRun.spec && latestRun.spec.params
+          ? latestRun.spec.params
+          : pipeline.spec && pipeline.spec.params
+            ? pipeline.spec.params
+            : null,
+      trigger: {
+        type: 'manual',
+      },
+    },
+  };
+};
+
+export const triggerPipeline = (
+  pipeline: Pipeline,
+  latestRun: PipelineRun,
+  redirectURL: string,
+): ActionFunction => {
+  //The returned function will be called using the 'kind' and 'obj' in Kebab Actions
+  return (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    label: 'Trigger',
+    callback: () => {
+      k8sCreate(PipelineRunModel, newPipelineRun(pipeline, latestRun)).then(() => {
+        if (redirectURL) {
+          redirectToResourceList(redirectURL);
+        }
+      });
+    },
+  });
+};
+
+export const reRunPipelineRun = (pipelineRun: PipelineRun): ActionFunction => {
+  //The returned function will be called using the 'kind' and 'obj' in Kebab Actions
+  return (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    label: 'Rerun',
+    callback: () => {
+      if (
+        !pipelineRun ||
+        !pipelineRun.metadata ||
+        !pipelineRun.metadata.namespace ||
+        !pipelineRun.spec ||
+        !pipelineRun.spec.pipelineRef ||
+        !pipelineRun.spec.pipelineRef.name
+      ) {
+        // eslint-disable-next-line no-console
+        console.error('Improper PipelineRun metadata');
+        return;
+      }
+      k8sList(PipelineRunModel, {
+        labelSelector: { 'tekton.dev/pipeline': pipelineRun.spec.pipelineRef.name },
+      }).then((listres) => {
+        k8sCreate(
+          PipelineRunModel,
+          newPipelineRun(
+            {
+              apiVersion: `${PipelineModel.apiGroup}/${PipelineModel.apiVersion}`,
+              kind: 'Pipeline',
+              metadata: {
+                name: pipelineRun.spec.pipelineRef.name,
+              },
+            },
+            {
+              ...pipelineRun,
+              metadata: {
+                ...pipelineRun.metadata,
+                name: getLatestRun({ data: listres }, 'creationTimestamp').metadata.name,
+              },
+            },
+          ),
+        );
+      });
+    },
+  });
+};
+
+export const rerunPipeline = (
+  pipeline: Pipeline,
+  latestRun: PipelineRun,
+  redirectURL: string,
```

Comment:
```suggestion
  redirectURL?: string,
```
So you can provide a null value instead of an empty string.

Reply:
done